### PR TITLE
ci: add checksums for release assets

### PR DESCRIPTION
**Pull request by shanewxy**
**Problem:**

Add checksum for release assets.

**Related Issue:**
#2057

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -196,6 +196,13 @@ jobs:
           remote_path: releases/${{ steps.metadata.outputs.version == 'main' && 'latest' || steps.metadata.outputs.version }}
           accelerate: true
           clean: false
+      - name: Generate SHA256 checksums
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          sha256sum .dist/package/${{ matrix.target }}/${{ matrix.task }}/build/cli/* > sha256sums.txt
+          sha256sum hack/mirror/walrus-images.txt >> sha256sums.txt
+          sha256sum hack/mirror/walrus-load-images.sh >> sha256sums.txt
+          sha256sum hack/mirror/walrus-save-images.sh >> sha256sums.txt
```
Comment (thxCode) [inline .github/workflows/ci.yml:205]: `sed` always includes a useless dirname, e.g. https://github.com/seal-io/terraform-provider-courier/blob/20612f54d1335c3b206ef3e519b7273384f4d8ff/hack/release.sh#L17, we need to trim them.

Comment (thxCode) [reply .github/workflows/ci.yml:205]: can we find a release action implementation with shasum generation?

Comment (shanewxy) [reply .github/workflows/ci.yml]: Updated to use https://github.com/wangzuo/action-release-checksums. There are other implementations like https://github.com/marketplace/actions/generate-checksum, but turns out to include path as well.

**Review by thxCode — APPROVED** [review]
LGTM, @gitlawr we need a RC version to verify this later.

**Comment by thxCode** [comment]
https://github.com/wangzuo/action-release-checksums/blob/8e38f587f6e4ea5d1b2e350be1c51c5d72c36934/index.js#L38-L54, it looks like the new action download the assets from remote which uploading recently by the previous step. I approve of this at present. I look forward to integrating into the `make package` step later.

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**